src/Plugin/WebformElement/NetsEasyPaymentElement.php

```diff
@@ -115,30 +116,113 @@ public function alterForm(array &$element, array &$form, FormStateInterface $for
+          'data-payment-error-message' => $this->t('An error has occurred. Please try again later.'),
           'data-create-payment-url' => Url::fromRoute("os2forms_payment.createPayment",
             [
               'amountToPay' => $amount_to_pay,
               'callbackUrl' => $callback_url,
             ])->toString(TRUE)->getGeneratedUrl(),
         ],
+        '#limit_validation_errors' => [],
+        '#element_validate' => [[get_class($this), 'validateOptions']],
       ];
-      $form['payment_reference_field'] = [
+      $form['os2forms_payment_payment_reference_field'] = [
         '#type' => 'hidden',
         '#name' => 'payment_reference_field',
       ];
     }
   }
 
   /**
-   * Get recipient elements.
+   * Validates the payment object.
    *
-   * @phpstan-return array<string, mixed>
+   * @param array<mixed> $element
+   *   Form element object.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state object.
+   *
+   * @return mixed
+   *   Returns validation results.
+   */
+  public static function validateOptions(array &$element, FormStateInterface $form_state): mixed {
+    $paymentHelper = \Drupal::service('Drupal\os2forms_payment\Helper\PaymentHelper');
+    $netsEasyController = \Drupal::service('Drupal\os2forms_payment\Controller\NetsEasyController');
+
+    $paymentId = $form_state->getValue('os2forms_payment_payment_reference_field');
+    if (!$paymentId) {
+      return $form_state->setError(
+        $element,
+        t('The form could not be submitted. Please try again.')
+      );
+    }
+
+    $endpoint = $paymentHelper->getTestMode()
+      ? 'https://test.api.dibspayment.eu/v1/payments/' . $paymentId
+      : 'https://api.dibspayment.eu/v1/payments/' . $paymentId;
+
+    $paymentValidated = $netsEasyController->validatePayment($endpoint);
+
+    if (!$paymentValidated) {
+      return $form_state->setError(
+        $element,
+        t('The payment could not be validated. Please try again.')
+      );
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Modifies how the payment details is displayed under "Resultater".
+   *
+   * @param array<mixed> $element
+   *   Form element object.
+   * @param \Drupal\webform\WebformSubmissionInterface $webform_submission
+   *   Form submission object.
+   * @param array<mixed> $options
+   *   Result view details.
+   *
+   * @return mixed
+   *   Returns content for the results view.
+   */
+  protected function formatHtmlItem(array $element, WebformSubmissionInterface $webform_submission, array $options = []): mixed {
+    $payment_element_data = $webform_submission->getData()[$element['#webform_key']] ?? NULL;
+
+    if ($payment_element_data) {
+      $payment_data = json_decode($payment_element_data)->paymentObject ?? NULL;
+      if ($payment_data) {
+        $form['payment_id'] = [
+          '#type' => 'item',
+          '#title' => $this->t('Betalings ID'),
+          '#markup' => $payment_data->payment_id ?? '{Tom}' ?: '{Tom}',
```

Comment:
What does “{Tom}” mean? Use `–` (en dash) (or 👻) for a missing value. Or a translatable text, but using 👻 is more fun.